src/main/java/picard/sam/DownsampleSam.java

```diff
@@ -115,10 +122,12 @@ protected int doWork() {
         final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(in.getFileHeader(), true, OUTPUT);
         final ProgressLogger progress = new ProgressLogger(log, (int) 1e7, "Wrote");
         final DownsamplingIterator iterator = DownsamplingIteratorFactory.make(in, STRATEGY, PROBABILITY, ACCURACY, RANDOM_SEED);
+        final QualityYieldMetricsCollector metricsCollector = new QualityYieldMetricsCollector(true, false, false);
 
         while (iterator.hasNext()) {
             final SAMRecord rec = iterator.next();
             out.addAlignment(rec);
+            if (METRICS_FILE != null) metricsCollector.acceptRecord(rec, null);
```

Comment:
Agreed, making it of type `Optional<QualityYieldMetricsCollector>` is syntactic sugar :/